include/arch/arm/cortex_m/irq.h

```diff
@@ -153,8 +153,19 @@ extern void z_irq_spurious(void *unused);
 extern void _isr_wrapper(void);
 #endif
 
+/**
```

Comment:
What we really need to decide for the API, is;
- can be called by interrupts?
- should it be thread-safe or don't care?
-

Reply:
> can be called by interrupts?

yes

> should it be thread-safe or don't care?

I would say don't care. Are we afraid that interrupt priority level will change after that call? How likely is that?

> could it be generalized for all ARCHEs?

when use case comes some will do that. I wouldn't like to see PR pending on other platforms implementation when current use case is for peripheral driver on arm cortex.

Reply:
I agree that this should be called by interrupts. We shall document that.
Also, it is fine if this is an arm-only API: we call the function z_arm_xxx and we can use a z_arch_ static inline function in include/arch/arm.

Reply:
>I would say don't care. Are we afraid that interrupt priority level will change after that call? How likely is that?

Now, this is problematic. If we make an API that checks if we can be preempted, we need to make sure this works in a deterministic way. I thought about it, and I don't want this to be fuzzy. We need to make this thread-safe, right?